src/core-asyncpipeline/AsyncPipeline/AsyncPipeline.T/AsyncPipeline.T.cs

```diff
@@ -0,0 +1,27 @@
+#nullable enable
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System
+{
+    public readonly partial struct AsyncPipeline<T> : IEquatable<AsyncPipeline<T>>
+    {
+        private static readonly CancellationToken CanceledToken;
```

Comment:
I suppose `CanceledToken` static field wants to be remade as `CanceledToken()` static method factory for the following reasons:
- In the scope of the pipeline instance it targeted to call only once (in conversion to a task), and the static field looks redundant.
- Moreover, it looks like some specific object (related to the underlying system) should not be instantiated in the static scope, and also it should not be shared between pipelines of the same T - for the reliability reasons.

Reply:
Done